docs/Architectural-Overview.md

```diff
@@ -125,7 +130,166 @@ Its a REST service which provides APIs to stores build wise test results and agg
+
+ 
+
+ 
+
+ 
+
+ 
+
+ 
+
+## Framework Deployment Flow: 
+
+This framework can be deployed and used in two ways. In multi auto bot setup mode and single auto bot setup mode. 
+
+ 
+
+## Multi Auto bot's setup: 
+
+ 
+
+ 
+
+sequence 1 to 6 is explained below in points. 
+
+Jenkins/Scheduler triggers a job which supplies parameters like targets and config to distributed executor i.e., master autobot. 
+
+Master Autobot runs in distributed mode and creates an execution plan which determines the sequence of test execution and which tests can be run in parallel. This execution plan entries are feed to Kafka topic cortx-test-jobs in JSON format. 
+
+Test runners (single autobot) are running in distributed mode on one or more virtual machines which have Kafka consumer. They belong to the same consumer group, and it guaranteed that a test will be executed only once by any of the test runners. 
+
+The deserialized JSON is parsed, and test metadata is sent to autobots (using pytest runner) to run either test with a parallel tag in parallel or other tests in serial order. Autobot also decides the target (Setup) to run tests against it from parsed JSON. 
+
+Multiple autobots can point to the same target. Optimal utilization of target is decided on tags or markers attached to a test function. e.g., marker with non-destructive tag can run in parallel across autobots deployed on different virtual machines/ containers. Target Locking section in Framework components explains about how target is utilized for efficient test execution. 
+
+Test reporting data is saved by Restful Reporting Server to Mongo DB through Reporting client API. Reporting Client is called at the end of pytest test run with pytest reporting hook. This data is retrieved from Reporting server for further analytics and presentation. 
+
+ 
+
+ 
+
+# Single Auto bot's setup: 
+
+This setup is useful when there is only one target. It is designed to be used by developers for their test validation/development. This setup uses a single autobot. 
+
+ 
+
+ 
+
+ 
+
+ 
+
+ 
+
+ 
+
+ 
+
+ 
+
+# Framework at system level: 
+
+Below diagram shows how this framework resides in any system and shows it’s use cases to different applications. 
+
+ 
+
+## Benefits: 
+
+Can run on commodity H/W and light weight VMs. The run can be started on a single client VM for multiple targets. (Storage efficient) 
+
+Framework will start parallel execution on multiple targets; depending on health of target, it will reschedule test execution for errored or failed tests. This will ensure complete test execution of identified test plan.  
+
+Framework will have parallel execution over a single target also which will reduce the time required for test execution. (Time efficient) 
+
+Maximum failures will get during the first few hours of execution. 
```

Comment:
Maybe we can reword it to "Parallel execution will reduce the time required to uncover failures during testing"